src/socket/ip_socket.cr

```diff
@@ -12,7 +20,15 @@ class IPSocket < Socket
     IPAddress.from(sockaddr, addrlen)
   end
 
+  # Returns the `IPAddress` for the remote end of the IP socket or `nil` if it
+  # is not connected.
+  def remote_address?
```

Comment:
Is this really needed?

Reply:
I felt that I needed it at some point during refactoring the specs. It's not completely thought out, but it came in handy.
But it's certainly up for debate.

Reply:
Maybe it's even better to remove it from this PR and discuss it separately.